mmv1/third_party/terraform/tests/resource_bigquery_table_test.go

```diff
@@ -11,6 +11,235 @@ import (
+			ExpectDiffSuppress: false,
+		},
+		"mode NULLABLE -> default mode (also NULLABLE)": {
+			Old: `[
+				{
+					"mode": "NULLABLE",
+					"name": "PageNo",
+					"type": "INTEGER"
+				}
+			]`,
+			New: `[
+				{
+					"name": "PageNo",
+					"type": "INTEGER"
+				}
+			]`,
+			ExpectDiffSuppress: true,
+		},
+		"type INTEGER -> INT64": {
+			Old:                "[{\"name\": \"someValue\", \"anotherKey\" : \"anotherValue\", \"type\": \"INTEGER\"  }]",
+			New:                "[{\"name\": \"someValue\", \"anotherKey\" : \"anotherValue\", \"type\": \"INT64\"  }]",
+			ExpectDiffSuppress: true,
+		},
+		"type INTEGER -> other": {
+			Old:                "[{\"name\": \"someValue\", \"anotherKey\" : \"anotherValue\", \"type\": \"INTEGER\"  }]",
+			New:                "[{\"name\": \"someValue\", \"anotherKey\" : \"anotherValue\", \"type\": \"somethingRandom\"  }]",
+			ExpectDiffSuppress: false,
+		},
+		"type FLOAT -> FLOAT64": {
+			Old:                "[{\"name\": \"someValue\", \"anotherKey\" : \"anotherValue\", \"type\": \"FLOAT\"  }]",
+			New:                "[{\"name\": \"someValue\", \"anotherKey\" : \"anotherValue\", \"type\": \"FLOAT64\"  }]",
+			ExpectDiffSuppress: true,
+		},
+		"type FLOAT -> default": {
+			Old:                "[{\"name\": \"someValue\", \"anotherKey\" : \"anotherValue\", \"type\": \"FLOAT\"  }]",
+			New:                "[{\"name\": \"someValue\", \"anotherKey\" : \"anotherValue\" }]",
+			ExpectDiffSuppress: false,
+		},
+		"type BOOLEAN -> BOOL": {
+			Old:                "[{\"name\": \"someValue\", \"anotherKey\" : \"anotherValue\", \"type\": \"BOOLEAN\"  }]",
+			New:                "[{\"name\": \"someValue\", \"anotherKey\" : \"anotherValue\", \"type\": \"BOOL\"  }]",
+			ExpectDiffSuppress: true,
+		},
+		"type BOOLEAN -> default": {
+			Old:                "[{\"name\": \"someValue\", \"anotherKey\" : \"anotherValue\", \"type\": \"BOOLEAN\"  }]",
+			New:                "[{\"name\": \"someValue\", \"anotherKey\" : \"anotherValue\" }]",
+			ExpectDiffSuppress: false,
+		},
+		"reordering fields": {
+			Old: `[
+				{
+					"name": "PageNo",
+					"type": "INTEGER"
+				},
+				{
+					"name": "IngestTime",
+					"type": "TIMESTAMP"
+				}
+			]`,
+			New: `[
+				{
+					"name": "IngestTime",
+					"type": "TIMESTAMP"
+				},
+				{
+					"name": "PageNo",
+					"type": "INTEGER"
+				}
+			]`,
+			ExpectDiffSuppress: true,
+		},
+		"reordering fields with value change": {
+			Old: `[
+				{
+					"name": "PageNo",
+					"type": "INTEGER",
+					"description": "someVal"
+				},
+				{
+					"name": "IngestTime",
+					"type": "TIMESTAMP"
+				}
+			]`,
+			New: `[
+				{
+					"name": "IngestTime",
+					"type": "TIMESTAMP"
+				},
+				{
+					"name": "PageNo",
+					"type": "INTEGER",
+					"description": "otherVal"
+				}
+			]`,
+			ExpectDiffSuppress: false,
+		},
+		"nested field ordering changes": {
+			Old: `[
+				{
+					"name": "someValue",
```

Comment:
For the sake of completeness, can we reorder an additional top-level field in this example?

Alternatively, adding a "complete" example that:

* reorders top level fields
* reorders nested fields
* includes a policy_tags value on both a top level and nested field

could be a good idea- one of the problems with the `schema` field is that we haven't captured enough complexity in the tests, historically.

Reply:
sure, done